internal/api/client/streaming/stream.go

```diff
@@ -153,25 +154,22 @@ var pingMsg = []byte("ping!")
 //			description: bad request
 func (m *Module) StreamGETHandler(c *gin.Context) {
 	var (
-		token         string
-		tokenInHeader bool
-		account       *gtsmodel.Account
-		errWithCode   gtserror.WithCode
+		account     *gtsmodel.Account
+		errWithCode gtserror.WithCode
 	)
 
-	if t := c.Query(AccessTokenQueryKey); t != "" {
-		// Token was provided as
-		// query param, no problem.
-		token = t
-	} else if t := c.GetHeader(AccessTokenHeader); t != "" {
-		// Token was provided in "Sec-Websocket-Protocol" header.
-		//
-		// This is hacky and not technically correct but some
-		// clients do it since Mastodon allows it, so we must
-		// also allow it to avoid breaking expectations.
-		token = t
-		tokenInHeader = true
-	}
+	// Check both query parameter AND header "Sec-Websocket-Protocol"
+	// value for a token. The latter is hacky and not technically
+	// correct, but some client do it since Mastodon allows it, so
+	// we must allow it for compatibility.
+	//
+	// Chrome also *always* expects the "Sec-Websocket-Protocol"
+	// response value to match input, so it must always be checked.
+	queryToken := c.Query(AccessTokenQueryKey)
+	headerToken := c.Query(AccessTokenHeader)
```

Comment:
Should it be `GetHeader()`?
```suggestion
	headerToken := c.GetHeader(AccessTokenHeader)
```

Reply:
indeed, thank you :sweat_smile: